extensions/_beacons/http_status.py

```diff
@@ -40,33 +46,43 @@ def validate(config):
     '''
     Validate the beacon configuration
     '''
+    valid = True
+    messages = []
+
     if not isinstance(config, list):
-        return False, ('Configuration for %s beacon must '
-                       'be a list.', config)
+        valid = False
+        messages.append('[-] Configuration for %s beacon must be a list', config)
     else:
         _config = {}
         list(map(_config.update, config))
 
     try:
         sites = _config.get('sites', {})
     except AttributeError:
-        return False, ('Sites for %s beacon '
-                       'must be a dict.', __virtualname__)
+        valid = False
+        messages.append('[-] Sites for %s beacon must be a dict', __virtualname__)
+
     if not sites:
-        return False, ('You neglected to define any sites')
+        valid = False
+        messages.append('[-] Configuration does not contain sites')
 
     for site, settings in sites.items():
         if required_site_attributes.isdisjoint(set(settings.keys())):
-            return False, ('Sites for {} beacon requires {}'.format(__virtualname__,
-                                                                    required_site_attributes))
-        try:
-            if optional_site_attributes.isdisjoint(set(settings.keys())):
-                return False, ('Sites for {} beacon requires {}'.format(__virtualname__,
-                                                                        optional_site_attributes))
-        except NameError:
-            log.info('No optional attributes defined')
+            valid = False
+            messages.append('[-] Sites for {} beacon requires {}'.format(__virtualname__,
+                                                                         required_site_attributes))
+        log.debug('[+] site: %s', site)
+        log.debug('[+] settings: %s', settings)
+
+        for optional_attrs in itertools.chain(settings.get(attr, []) for attr in optional_site_attributes):
+            for item in optional_attrs:
+                cmp = item.get('comp')
+                if cmp and not cmp in comparisons:
+                    valid = False
+                    messages.append('Invalid comparison operator %s', cmp)
```

Comment:
Looks like you've got a convention of prepending `[-]` for the messages which is absent here

Reply:
Changed